### PR TITLE
adding robot/training notebooks

### PR DESCRIPTION
**Pull request by paularamo**
# Description

- Provide a summary of the modification as well as the issue that has been resolved. List any dependencies that this modification necessitates.

- Fixes # (issue)

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

**Comment by paularamo** [comment]
image

The heat map looks weird. Apart of that, the two notebooks are working well. 501_1 is working pretty well with the robot setup.

**Review by ashwinvaidya17 — COMMENTED** [review]
I am in favour of the following convention for the notebooks and the dobot folder
```bash
notebooks/500_use_cases
└── 501_Dobot
    ├── 501a_Dataset_creation_and_Inference_with_a_robotic_arm.ipynb
    ├── 501b_Training_a_model_with_cubes_from_a_robotic_arm.ipynb
```
There are some issues in the feature branch due to which the CI is failing.

File: notebooks/500_use_cases/501_dobot/cubes_config.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Replace the dataset configs with the following.
+dataset:
+  name: cubes
+  format: folder
+  path: ./datasets/cubes
```
Comment (ashwinvaidya17) [inline notebooks/500_use_cases/501_dobot/cubes_config.yaml:5]:
```suggestion
  path: /datasets/cubes
```

File: notebooks/500_use_cases/501_dobot/cubes_config.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  overfit_batches: 0.0
+  plugins: null
+  precision: 32
+  profiler: null
+  reload_dataloaders_every_n_epochs: 0
+  replace_sampler_ddp: true
+  sync_batchnorm: false
+  tpu_cores: null
+  track_grad_norm: -1
+  val_check_interval: 1.0 # Don't validate before extracting features.
```
Comment (ashwinvaidya17) [inline notebooks/500_use_cases/501_dobot/cubes_config.yaml:120]:
```suggestion
  val_check_interval: 1.0 # Don't validate before extracting features.

```

File: .gitignore [hunk 3]
```diff
@@ -6,8 +6,6 @@ results
 !anomalib/core/results
 
 # Jupyter Notebooks
```
Comment (ashwinvaidya17) [inline .gitignore:8]:
```suggestion

```

File: notebooks/500_use_cases/501_dobot/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+       
+Step 4: Then connect your USB Camera and verify it works using a simple camera application. Once it is verified, close the application. 
+
+
+Step 5 (Optional): If you have the Dobot robot please make the following.
+
+1.  Install Dobot requirements (See Dobot documentation here: https://en.dobot.cn/products/education/magician.html).
+2.  Check all connections to the Dobot and verify it is working using the Dobot Studio.
+3.  Install the vent accessory on the Dobot and verify it is working using Dobot Studio.
+4.  In the Dobot Studio, hit the "Home" button, and locate the:
```
Comment (ashwinvaidya17) [inline notebooks/500_use_cases/501_dobot/README.md:36]:
```suggestion
1. Install Dobot requirements (See Dobot documentation here: https://en.dobot.cn/products/education/magician.html).
2. Check all connections to the Dobot and verify it is working using the Dobot Studio.
3. Install the vent accessory on the Dobot and verify it is working using Dobot Studio.
4. In the Dobot Studio, hit the "Home" button, and locate the:
```

**Comment by ashwinvaidya17** [comment]
I am merging this to the feature branch. I'll create a new PR which addressed the linting issues.

**Pull request merged**